src/Libraries/RevitNodesUI/Selection.cs

```diff
@@ -108,6 +148,12 @@ public override void Dispose()
             RevitServicesUpdater.Instance.ElementsDeleted -= Updater_ElementsDeleted;
             RevitServicesUpdater.Instance.ElementsModified -= Updater_ElementsModified;
             DocumentManager.Instance.CurrentUIApplication.Application.DocumentOpened -= Controller_RevitDocumentChanged;
+
+            if (null != revitDynamoModel)
+            {
+                var hwm = RevitDynamoModel.Workspaces.OfType<HomeWorkspaceModel>().ElementAt(0);
+                hwm.RunSettings.PropertyChanged -= revMod_PropertyChanged;
+            }
```

Comment:
What about opening a new file (which will update `HomeWorkspaceModel` inside `DynamoModel`)? In that case we need to unregister from the event. If that is the case, we might want to detect property change of `CurrentSpace` on `DynamoModel`, and subscribe/unsubscribe the handler.